### PR TITLE
use SDK functions for decoding & encoding uint64

### PR DESCRIPTION
**Pull request by ratik21**

### CODE REVIEW

**Comment by amityadav0** [comment]
https://discord.com/channels/754286650540032030/754286650540032033/854032882778767370

File: packages/algob/test/lib/ssc.ts [hunk 1]
```diff
@@ -56,10 +56,9 @@ describe("Convert integer to big endian", () => {
   });
 
   it("should throw error if value is not uint64", () => {
-    let errMsg = "Invalid uint64 -5";
-    assert.throws(() => uint64ToBigEndian(MIN_UINT64 - 5n), errMsg);
+    const errMsg = "Input is not a 64-bit unsigned integer";
 
-    errMsg = "Invalid uint64 18446744073709551620";
+    assert.throws(() => uint64ToBigEndian(MIN_UINT64 - 5n), errMsg);
```
Comment (amityadav0) [inline packages/algob/test/lib/ssc.ts:61]: does sdk functions don't throw error ? maybe we can add a check before. 
this seems more clear `Invalid uint64 18446744073709551620` (in terms of debugging also it is good)

File: packages/runtime/src/types.ts [hunk 2]
```diff
@@ -429,3 +429,13 @@ export interface AssetModFields {
   freeze?: string
   clawback?: string
 }
+
+/**
+ * SDK decoding types (Configure how the integer will be decoded)
+ * https://github.com/algorand/js-algorand-sdk/blob/develop/src/encoding/uint64.ts#L29
+ */
+export enum DecodingMode {
```
Comment (amityadav0) [inline packages/runtime/src/types.ts:437]: how about moving these types-sdk ?

File: packages/algob/test/lib/ssc.ts (hunk 1 above)
Comment (ratik21) [reply packages/algob/test/lib/ssc.ts:61]: yes sdk function throws error. ok, adding check

File: packages/runtime/src/types.ts (hunk 2 above)
Comment (ratik21) [reply packages/runtime/src/types.ts:437]: yes

**Comment by ratik21** [comment]
> https://discord.com/channels/754286650540032030/754286650540032033/854032882778767370

https://discord.com/channels/754286650540032030/754286650540032033/854051475733872660

File: packages/types-algosdk/types/index.d.ts [hunk 3]
```diff
@@ -352,6 +352,35 @@ declare module 'algosdk' {
...
   export function decodeUnsignedTransaction(b: Uint8Array): Transaction;
 
+  /**
+   * encodeUint64 converts an integer to its binary representation.
+   * @param num - The number to convert. This must be an unsigned integer less than
+   *   2^64.
+   * @returns An 8-byte typed array containing the big-endian encoding of the input
+   *   integer.
+   */
+  export function encodeUint64(num: number | bigint): Uint8Array;
```
Comment (amityadav0) [inline packages/types-algosdk/types/index.d.ts:362]: let's add a link here to js-sdk code.

**Review by amityadav0 — APPROVED** [review]

**Pull request merged**